pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	// Opening new db connection
+	newDBConn, err := sql.Open("mysql", p.dbURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new connection %s: %w", p.dbURL, err)
+	}
+
+	// Use is used to select the created database without this DDL operations are not permitted
+	_, err = newDBConn.Exec(useDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to use db %s: %w", name, err)
+	}
+
+	tableName := tablePrefix + name
+	// key has max varchar size it can accommodate as per mysql 8.0 spec
+	createTableStmt := "CREATE Table IF NOT EXISTS " + tableName +
+		"(`key` varchar(255) NOT NULL ,`value` BLOB, PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	_, err = newDBConn.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        newDBConn,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errors.New("key is mandatory")
+	}
+
+	//nolint: gosec
+	// create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errors.New("key is mandatory")
+	}
+
+	var value []byte
+	//nolint: gosec
+	// select query to fetch the record by key
+	err := s.db.QueryRow("SELECT `value` FROM "+s.tableName+" "+
+		" WHERE `key` = ?", k).Scan(&value)
+	if err != nil {
+		if strings.Contains(err.Error(), sqlDBNotFound) {
+			return nil, storage.ErrValueNotFound
+		}
+
+		return nil, fmt.Errorf("failed to get row %w", err)
+	}
+
+	return value, nil
+}
+
+type sqlDBResultsIterator struct {
+	resultRows *sql.Rows
+	err        error
+}
+
+func (s *sqlDBStore) CreateIndex(createIndexRequest storage.CreateIndexRequest) error {
+	// TODO: Issue-36 Implement Create Index
+	return nil
+}
```

Comment:
In memstore.go you'll find

`var errIndexingNotSupported = errors.New("indexing is not supported in memstore")
var errQueryingNotSupported = errors.New("querying is not supported in memstore")`

These are used for the `edv` project to determine when an underlying store doesn't support indexing. You should move these into `store.go`, change the error message to be more generic (remove "memstore" wording) and then also return the same error in CreateIndex here.

(also - can you make them public? This will help me in updating the EDV project after this)